packages/langium/src/grammar/langium-grammar-goto.ts

```diff
@@ -0,0 +1,47 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { findParentParserRuleCstNode, isAssignment, isInterface, isParserRule } from '..';
+import { DefaultGoToResolverProvider } from '../lsp';
+import { LangiumServices } from '../services';
+import { CstNode } from '../syntax-tree';
+import { getDocument } from '../utils/ast-util';
+import { LangiumDocument } from '../workspace/documents';
+
+export class LangiumGrammarGoToResolver extends DefaultGoToResolverProvider {
+
+    constructor(services: LangiumServices) {
+        super(services);
+    }
+
+    protected findTargetNode(sourceCstNode: CstNode, targetCstNodes: Array<{ source: CstNode; target: CstNode; targetDocument: LangiumDocument; }>): void {
+        if (isAssignment(sourceCstNode.element)){
+            const parserRule = findParentParserRuleCstNode(sourceCstNode);
+            // check that the parser rule has a return type
+            if (parserRule && isParserRule(parserRule.element) && parserRule.element.returnType) {
```

Comment:
Suggestion: I believe that this can be greatly simplified. You can just use `getContainerOfType(sourceCstNode.element, isParserRule)` to get the parent parser rule of the assignment. I think that eliminates the need for `findParentParserRuleCstNode` as well.